### ISSUE TITLE
[Bug?]: Yarn install deletes existing empty node_modules directory

### ISSUE THREAD

**Issue by mirogta** [yarnpkg/berry#3672]
### Self-service

- [ ] I'd be willing to implement a fix

### Describe the bug

**Actual Behaviour:**

Given I'm using yarn 3.1.0
And I'm using `nodeLinker: node-modules`
And I have an existing repository
And I have mounted `node_modules` directory to a RAM drive (with `ln -s /ramdrive/node_modules node_modules` 
 following https://dev.to/llldar/speed-up-your-build-by-putting-nodemodules-in-ram-15h4)
And the `node_modules` directory is empty
When I run `yarn install`
Then the existing empty `node_modules` directory is deleted
And a new `node_modules` directory is recreated, but not mounted to a RAM drive obviously
And the dependencies are installed there

**Expected Behaviour:**
…
Then the existing empty `node_modules` directory is kept (not deleted)
And the dependencies are installed there

**Additional Notes:**

Possibly related to #5709

I understand that everything *inside* `node_modules` should be managed exclusively by yarn. That's OK, but why does it need to delete and recreate the `node_modules` directory when it's actually empty?

### To reproduce

1. Take any repository using yarn2+ and `nodeLinker: node-modules`
2. Create a RAM drive
3. Create `node_modules` directory in the RAM drive, e.g. `/ramdrive/node_modules`
4. Mount it inside your repository, e.g. `ln -s /ramdrive/node_modules node_modules`
5. Run `ln -ls` and observe the `node_modules` is a symbolic link pointing to the path on the RAM drive
6. Run `yarn install`
7. Run `ln -ls` and observe the `node_modules` is now a normal directory, because the symbolic link has been deleted and a new directory has been created

### Environment

```shell
System:
    OS: macOS 11.6
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
  Binaries:
    Node: 16.11.1 - /private/var/folders/bn/28sdpc555wg0ffgr2zd6c5tw0000gn/T/xfs-1c10d4e3/node
    Yarn: 3.0.2 - /private/var/folders/bn/28sdpc555wg0ffgr2zd6c5tw0000gn/T/xfs-1c10d4e3/yarn
    npm: 8.0.0 - /usr/local/bin/npm
```

**Comment by mirogta** [comment]
@larixer I've tried to set the yarn version to the branch #2487 version but that has failed with this error:

```
$ yarn set version from sources --branch 2487
➤ YN0000: Fetching the latest commits

  $ git fetch origin pull/2487/head --force
  $ git reset --hard FETCH_HEAD
  $ git clean -dfx

➤ YN0000: Building a fresh bundle

Usage Error: Unrecognized or legacy configuration settings found: changesetIgnorePatterns - run "yarn config -v" to see the list of settings supported in Yarn (in /private/var/folders/bn/28sdpc555wg0ffgr2zd6c5tw0000gn/T/yarnpkg-sources/d0a670/.yarnrc.yml)

$ yarn run [--inspect] [--inspect-brk] <scriptName> ...
➤ YN0001: Error: Child "yarn" exited with exit code 1
    at ChildProcess.<anonymous> (/Users/miro/.yarn/releases/yarn-3.0.2.cjs:177:63920)
    at ChildProcess.emit (node:events:390:28)
    at maybeClose (node:internal/child_process:1064:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:301:5)
➤ YN0000: Failed with errors in 1s 258ms
```

**Comment by mirogta** [comment]
I can confirm that this issue has been fixed in `yarn` version 3.1.1

Thank you!
